import-service/lib/import-service-stack.ts

```diff
@@ -0,0 +1,130 @@
+
+export class ImportServiceStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props)
+
+    // Create S3 bucket for file uploads
+    const bucket = new s3.Bucket(this, "slava-s3-bucket", {
+      cors: [
+        {
+          allowedMethods: [
+            s3.HttpMethods.GET,
+            s3.HttpMethods.PUT,
+            s3.HttpMethods.POST,
+            s3.HttpMethods.DELETE,
+            s3.HttpMethods.HEAD,
+          ],
+          allowedOrigins: ["*"],
+          allowedHeaders: ["*"],
+          exposedHeaders: [
+            "ETag",
+            "x-amz-server-side-encryption",
+            "x-amz-request-id",
+            "x-amz-id-2",
+          ],
+        },
+      ],
+    })
+
+    // Create Lambda function using NodejsFunction
+    const importProductsFile = new NodejsFunction(this, "ImportProductsFile", {
+      runtime: cdk.aws_lambda.Runtime.NODEJS_18_X,
+      depsLockFilePath: require.resolve('../package.json'),
+      handler: "handler",
+      entry: path.join(__dirname, "../src/importProductsFile.ts"),
+      environment: {
+        BUCKET_NAME: bucket.bucketName,
+      },
+      // bundling: {
+      //   minify: true,
+      //   sourceMap: true,
+      //   externalModules: ['aws-sdk'],
+      // },
+    })
+
+    // Add explicit S3 permissions
+    importProductsFile.addToRolePolicy(
+      new iam.PolicyStatement({
+        actions: ["s3:PutObject", "s3:GetObject"],
+        resources: [`${bucket.bucketArn}/*`, bucket.bucketArn],
+      })
+    )
+
+    // Create importFileParser Lambda
+    const importFileParser = new NodejsFunction(this, "ImportFileParser", {
+      runtime: lambda.Runtime.NODEJS_18_X,
+      depsLockFilePath: require.resolve('../package.json'),
+      handler: "handler",
+      entry: path.join(__dirname, "../src/importFileParser.ts"),
+      environment: {
+        BUCKET_NAME: bucket.bucketName,
+      },
+      // bundling: {
+      //   minify: true,
+      //   sourceMap: true,
+      //   externalModules: ['aws-sdk'],
+      // },
+      timeout: cdk.Duration.seconds(60), // Increase timeout for file processing
+    })
+
+    // Grant S3 permissions to Lambda
+    bucket.grantReadWrite(importProductsFile)
+    bucket.grantReadWrite(importFileParser)
+    bucket.grantDelete(importFileParser)
+
+    // Add S3 notification for uploaded folder
+    bucket.addEventNotification(
+      s3.EventType.OBJECT_CREATED,
+      new s3n.LambdaDestination(importFileParser),
+      { prefix: "uploaded/" } // Only trigger for objects in uploaded folder
+    )
+
+    // Create API Gateway
+    const api = new apigateway.RestApi(this, "ImportApi", {
+      defaultCorsPreflightOptions: {
+        allowOrigins: apigateway.Cors.ALL_ORIGINS,
+        allowMethods: apigateway.Cors.ALL_METHODS,
+      },
+    })
+
+    // Create API Gateway integration
+    const integration = new apigateway.LambdaIntegration(importProductsFile, {
```

Comment:
redundant variable assign.

```suggestion
    new apigateway.LambdaIntegration(importProductsFile, {
```

Reply:
I'm using it here:

```
importResource.addMethod("GET", integration, {
      requestParameters: {
        "method.request.querystring.name": true,
      },
```